modules/datastore/src/Controller/AbstractQueryController.php

```
@@ -221,6 +221,9 @@ protected function extractMetastoreDependencies(DatastoreQuery $datastoreQuery):
   protected function buildDatastoreQuery(Request $request, mixed $identifier = NULL) {
```

`\Drupal\datastore\Controller\QueryDownloadController::buildDatastoreQuery()` overrides this... Both call `getPayloadJson()` which is probably where we should throw the exception.

So: Move this checking to either `getPayloadJson()` or `fixTypes()`, and add a bad JSON test for both query controller and download query controller.

---

Done